packages/as-types/assembly/argument.ts

```diff
@@ -243,8 +244,20 @@ export class Args {
     return new Result(!!this.serialized[this.offset++]);
   }
 
+  /**
+   * This function deserialize an object by calling its `deserialize` method.
+   *
+   * @param object - the object to deserialize
+   * @returns the object
+   */
+  nextObject<T extends Serializable<T>>(object: T): T {
```

Comment:
I'm not sure the terminology is weel choosen here. If I understand well, you will use your code like that:

``` typescript
const c2 = new Currency();
const useless = args.nextObject(c2);
```
If i'm right, I'm proposing two things:
- remove the useless returned value
- rename your function nextHydrate

hence, the usage code would be:
```typescript
const c2 = new Currency();
args.nextHydrate(c2);
```

Reply:
I see your point, but you can do:

```typescript
const deserialized = args.nextObject<Address>(new Address());
```

what does hydrate mean?

Reply:
Ok. then you can ignore my comment :-)

Reply:
I don't understand neither what "hydrate" means in that case...
I only know "Hydrate" term used to define the action of feeding a template with values...
My proposition on this would be "nextSerializable"

Reply:
I like nextSerializable